web/db/actions/Post.ts

```diff
@@ -82,6 +90,57 @@ async function getResizedUploadUrl(uuid: string): Promise<string> {
   return `${consts.baseUrl}/api/resizedUpload/${token}`;
 }
 
+async function deleteAttachments(keysToDelete: string[]) {
+  let numTries = 1;
+  const maxTries = 3;
+  const objectsToDelete = keysToDelete.map((keyToDelete) => ({
+    Key: keyToDelete,
+  }));
+  const deleteObjectsCommand = new DeleteObjectsCommand({
+    Bucket: consts.storageBucket,
+    Delete: { Objects: objectsToDelete, Quiet: true },
+  });
+  while (true) {
+    try {
+      const returned = await storageClient.send(deleteObjectsCommand);
+      if (!returned["Deleted"]) {
+        return { success: true };
+      }
```

Comment:
There's one thing I didn't catch previously - we need to check the length of the `Deleted` field to see if it matches `keysToDelete.length`. 

```js
if (returned.Deleted?.length === keysToDelete.length) {
  return { success: true };
} else {
  throw new Error("All attachments not successfully deleted.");
}
```

Reply:
The `Deleted` field only returns the elements that haven't been deleted. Also, previously in a previous comment, you had mentioned to go ahead with post deletion even if all the attachments hadn't been deleted (potentially writing a cron job to go through those). The way we've modified it now in around the last two commits, if it's a partial deletion we throw an error for the whole thing and do not proceed with post deletion. Is that okay?

Reply:
Yes, you're right. Completely forgot about quiet mode. Let's revert back to the default mode so we get the results of every key. I'm thinking that we should still throw an error from the database action but just add another catch in the try block:

`deleteAttachments` should stay like this:
```js
try {
  const returned = await storageClient.send(deleteObjectsCommand);
  if (returned.Deleted?.length === keysToDelete.length) {
    return { success: true };
  } else {
    throw new Error("All attachments not successfully deleted.");
  }
} catch (e) {
  //TODO: Write cron job to mark unsuccessful deletions to delete later
  if (numTries++ == maxTries) {
    throw e;
  }
}
```

`deletePost`:
```js
const post = await getPost(id, false);
>>> const returned = deleteAttachments(post.attachments).catch((e) => Sentry.captureException(e));
const deletePost = await Post.deleteOne({ _id: id });
if (deletePost.deletedCount !== 1) {
  throw new Error("Post document deletion failed");
}
```

Reply:
I removed the try-catch in `deletePost` since if there is an error here (that's not an error from `deleteAttachments`), it will get caught properly by `post.delete` the way you've handled it.

Reply:
So the way the `DeleteObjectsCommand` works with quiet mode false, it will return a list of every object under deleted whether successful or not. So `returned.Deleted?.length === keysToDelete.length` will always be true unless the `DeleteObjectsCommand` throws an error which it does not do for partial deletions or even no deletions and just for server errors etc. So to make sure that all attachments are deleted, probably the best way would be to use quiet mode and check if `returned["Deleted"]` exists as it's done right now. At least that's how I've seen the `DeleteObjectsCommand` work when I was figuring it out and it seems to match up with the docs at https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/classes/deleteobjectscommand.html .

Reply:
My understanding was that all the deleted items go into `Deleted`, the rest into `Errors` - I can get more clarifications on that

Reply:
See this: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-s3/Interface/DeleteObjectsCommandOutput/
<img width="820" alt="image" src="https://github.com/GTBitsOfGood/angels-among-us/assets/31021209/6b1e021b-2d61-45f3-a8f0-b6445af99306">

Reply:
Seems like `Deleted` can contain successfully deleted objects as well as objects that weren't found. We're not concerned about whether an object can be found or not - we want to pay attention to errors like _Access Denied_, _Forbidden_, S3 authorization issues, stuff like that.